python/console/console_editor.py

```diff
@@ -358,36 +364,34 @@ def objectListEditor(self):
             listObj.show()
             self.parent.pc.objectListButton.setChecked(True)
 
-    def codepad(self):
-        import urllib.request
-        import urllib.parse
-        import urllib.error
-        listText = self.selectedText().split('\n')
-        getCmd = []
-        for strLine in listText:
-            getCmd.append(strLine)
-        pasteText = "\n".join(getCmd)
-        url = 'http://codepad.org'
-        values = {'lang': 'Python',
-                  'code': pasteText,
-                  'submit': 'Submit'}
+    def shareOnGist(self, is_public):
+        import requests
+        import json
+
+        ACCESS_TOKEN = self.settings.value("pythonConsole/accessTokenGithub", '')
+        URL = "https://api.github.com/gists"
+
+        headers = {'Authorization': 'token %s' % ACCESS_TOKEN}
+        params = {'scope': 'gist'}
+
+        path = self.parent.tw.currentWidget().path
+        filename = os.path.basename(path) if path else None
+        filename = filename if filename else "pyqgis_snippet.py"
+
+        selected_text = self.selectedText()
+        data = {"description": "Gist created by PyQGIS Console",
+                "public": is_public,
+                "files": {filename: {"content": selected_text}}}
         try:
-            response = urllib.request.urlopen(url, urllib.parse.urlencode(values))
-            url = response.read()
-            for href in url.split("</a>"):
-                if "Link:" in href:
-                    ind = href.index('Link:')
-                    found = href[ind + 5:]
-                    for i in found.split('">'):
-                        if '<a href=' in i:
-                            link = i.replace('<a href="', "").strip()
-            if link:
-                QApplication.clipboard().setText(link)
-                msgText = QCoreApplication.translate('PythonConsole', 'URL copied to clipboard.')
-                self.parent.pc.callWidgetMessageBarEditor(msgText, 0, True)
-        except urllib.error.URLError as e:
-            msgText = QCoreApplication.translate('PythonConsole', 'Connection error: ')
-            self.parent.pc.callWidgetMessageBarEditor(msgText + repr(e.args), 0, True)
+            res = requests.post(URL, headers=headers, params=params, data=json.dumps(data)).json()
```

Comment:
Could this be changed to use QNetworkRequest or QgsBlockingNetworkRequest? Then we'd get proper handling of proxies, logging of the request, etc

Reply:
(I can do that if you don't want to!)

Reply:
I can give it a try, thanks for the insight!